rotkehlchen/chain/evm/utils.py

```diff
@@ -0,0 +1,140 @@
+    from rotkehlchen.types import LP_TOKEN_AS_POOL_CONTRACT_ABIS
+
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+LP_TOKEN_AS_POOL_PROTOCOL_TO_ABI_NAME: Final[dict[str, 'LP_TOKEN_AS_POOL_CONTRACT_ABIS']] = {
+    VELODROME_POOL_PROTOCOL: 'VELO_V2_LP',
+    UNISWAP_PROTOCOL: 'UNISWAP_V2_LP',
+}
+MULTICALL_DEPLOYMENT_BLOCK = 7929876
+
+
+def lp_price_from_pool_contract(
+        evm_inquirer: 'EvmNodeInquirer',
+        token: EvmToken,
+        token_price_func: Callable,
+        token_price_func_args: list[Any],
+        block_identifier: BlockIdentifier,
+) -> Optional[Price]:
+    """
+    This works for any uniswap like LP token. It calculates the price for an LP token the contract
+    of which is also the contract of the pool it represents. For example velodrome or uniswap lp
+    tokens. The price is calculated like this:
+    price = (Total value of liquidity pool) / (Current supply of LP tokens)
+    We need:
+    - Price of token 0
+    - Price of token 1
+    - Pooled amount of token 0
+    - Pooled amount of token 1
+    - Total supply of pool token
+    """
+    address = token.evm_address
+    try:
+        abi_name = LP_TOKEN_AS_POOL_PROTOCOL_TO_ABI_NAME[token.protocol]  # the lp token's contract abi that will be used to query the info needed for lp token price calculation  # noqa: E501
+    except KeyError:
+        log.debug(
+            f'There is no suitable contract abi for protocol {token.protocol} '
+            f'of token {token.evm_address}. Cannot calculate price',
+        )
+        return None
+
+    contract = EvmContract(
+        address=address,
+        abi=evm_inquirer.contracts.abi(abi_name),
+        deployed_block=0,
+    )
+    methods = ('token0', 'token1', 'totalSupply', 'getReserves', 'decimals')
+    if isinstance(block_identifier, int) and block_identifier <= MULTICALL_DEPLOYMENT_BLOCK:
+        log.error(
+            f'No multicall contract at the block {block_identifier}. {abi_name} '
+            f'query failed. Should implement direct queries',
+        )
+        return None
+
+    try:
+        output = evm_inquirer.multicall(
+            require_success=True,
+            calls=[(address, contract.encode(method_name=method)) for method in methods],
+            block_identifier=block_identifier,
+        )
+    except (RemoteError, BlockchainQueryError) as e:
+        log.error(
+            f'Remote error calling multicall contract for {abi_name} '
+            f'token {token.evm_address} properties: {e!s}',
+        )
+        return None
+
+    # decode output
+    decoded = []
+    for (method_output, method_name) in zip(output, methods):
+        call_success = True
+        if call_success and len(method_output) != 0:
+            decoded_method = contract.decode(method_output, method_name)
+            if len(decoded_method) == 1:
+                # https://github.com/PyCQA/pylint/issues/4739
```

Comment:
This was supposed to explain a pylint ignore, if you check the link. I don't see the pylint ignore statement there anymore.

```suggestion
```

Reply:
yes, that was my question too. Maybe it has been fixed now although the linked issue is still open.

Reply:
no no the issue is not fixed. But at some point code changed, type:ignore was removed since it was not needed but the comment explaining why it was there was not removed.